mllib/src/main/scala/org/apache/spark/ml/optim/IterativelyReweightedLeastSquares.scala

```diff
@@ -89,7 +89,7 @@ private[ml] class IterativelyReweightedLeastSquares(
       val oldCoefficients = oldModel.coefficients
       val coefficients = model.coefficients
       BLAS.axpy(-1.0, coefficients, oldCoefficients)
-      val maxTolOfCoefficients = oldCoefficients.toArray.reduce { (x, y) =>
+      val maxTolOfCoefficients = oldCoefficients.toArray.foldLeft(0.0) { (x, y) =>
```

Comment:
nit: this could be 
````scala
val maxTol = oldCoefficients.foldLeft(math.abs(oldModel.intercept - model.intercept)) { (x, y) => 
  math.max(math.abs(x), math.abs(y))
}
````